### PR TITLE
Refactory code and cleanup.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ✔️

- [x] Implement ConnectionInterface::class.
- [x] Rename all files.
- [x] Remove dependencies tests `Yiisoft/Db/Tests'.
- [x] Fixed namespace dev psr4.

### CODE REVIEW

File: src/Connection/PgsqlConnection.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+use Yiisoft\Db\Command\Command;
+use Yiisoft\Db\Exception\Exception;
+use Yiisoft\Db\Exception\InvalidConfigException;
+use Yiisoft\Db\Exception\NotSupportedException;
+use Yiisoft\Db\Pgsql\Schema\PgsqlSchema;
+
+/**
+ * Database connection class prefilled for PGSQL Server.
+ */
+final class PgsqlConnection extends Connection
```
Comment (samdark) [inline src/Connection/PgsqlConnection.php:18]: That looks a bit over-namespaced. I know that it is in line with `db` package namespaces but still... could it be `Yiisoft\Db\Pgsql\Connection`?

File: src/Query/PgsqlQueryBuilder.php [hunk 2]
```diff
@@ -37,9 +36,10 @@
 use function preg_replace;
 use function reset;
 use function strpos;
+use function strrpos;
 use function version_compare;
 
-class QueryBuilder extends AbstractQueryBuilder
+final class PgsqlQueryBuilder extends QueryBuilder
```
Comment (samdark) [inline src/Query/PgsqlQueryBuilder.php:42]: Isn't namespace "Yiisoft\\Db\\**Pgsql**\\Query" enough to disambiguate it?

File: src/Schema/PgsqlColumnSchema.php [hunk 3]
```diff
@@ -8,15 +8,16 @@
...
 
 use function array_walk_recursive;
+use function in_array;
 use function is_array;
 use function is_string;
 use function json_decode;
 use function strtolower;
 
-class ColumnSchema extends AbstractColumnSchema
+final class PgsqlColumnSchema extends ColumnSchema
```
Comment (samdark) [inline src/Schema/PgsqlColumnSchema.php:20]: Isn't namespace enough to tell that the `ColumnSchema` belongs to PostgreSQL?

File: src/Schema/PgsqlTableSchema.php [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
 use Yiisoft\Db\Schema\TableSchema as AbstractTableSchema;
 
-class TableSchema extends AbstractTableSchema
+final class PgsqlTableSchema extends AbstractTableSchema
```
Comment (samdark) [inline src/Schema/PgsqlTableSchema.php:9]: Pgsql prefix seems to duplicate information that is already there in the namespace.

File: src/Connection/PgsqlConnection.php (hunk 1 above)
Comment (terabytesoftw) [reply src/Connection/PgsqlConnection.php]: Fixed.

File: src/Query/PgsqlQueryBuilder.php (hunk 2 above)
Comment (terabytesoftw) [reply src/Query/PgsqlQueryBuilder.php]: Fixed.

File: src/Schema/PgsqlColumnSchema.php (hunk 3 above)
Comment (terabytesoftw) [reply src/Schema/PgsqlColumnSchema.php]: FIxed.

File: src/Schema/PgsqlTableSchema.php (hunk 4 above)
Comment (terabytesoftw) [reply src/Schema/PgsqlTableSchema.php]: Fixed.

**Review by samdark — APPROVED** [review]

**Pull request merged**